crates/bevy_pbr/src/render_graph/forward_pipeline/forward.frag

```diff
@@ -24,41 +27,170 @@ layout(set = 1, binding = 0) uniform Lights {
     Light SceneLights[MAX_LIGHTS];
 };
 
-layout(set = 3, binding = 0) uniform StandardMaterial_albedo {
-    vec4 Albedo;
-};
+layout(set = 3, binding = 0) uniform StandardMaterial_albedo { vec4 Albedo; };
 
-# ifdef STANDARDMATERIAL_ALBEDO_TEXTURE
+#ifdef STANDARDMATERIAL_ALBEDO_TEXTURE
 layout(set = 3, binding = 1) uniform texture2D StandardMaterial_albedo_texture;
-layout(set = 3, binding = 2) uniform sampler StandardMaterial_albedo_texture_sampler;
-# endif
+layout(set = 3,
+       binding = 2) uniform sampler StandardMaterial_albedo_texture_sampler;
+#endif
+
+#ifndef STANDARDMATERIAL_UNLIT
+
+layout(set = 3, binding = 3) uniform StandardMaterial_pbr { vec2 pbr; };
+
+#define saturate(x) clamp(x, 0.0, 1.0)
+const float PI = 3.141592653589793;
+
+float pow5(float x) {
+    float x2 = x * x;
+    return x2 * x2 * x;
+}
+
+float getSquareFalloffAttenuation(float distanceSquare, float falloff) {
+    float factor = distanceSquare * falloff;
+    float smoothFactor = saturate(1.0 - factor * factor);
+    return smoothFactor * smoothFactor;
+}
+
+float getDistanceAttenuation(const vec3 posToLight, float falloff) {
+    float distanceSquare = dot(posToLight, posToLight);
+    float attenuation = getSquareFalloffAttenuation(distanceSquare, falloff);
+    return attenuation * 1.0 / max(distanceSquare, 1e-4);
+}
+
+float D_GGX(float roughness, float NoH, const vec3 h) {
+    float oneMinusNoHSquared = 1.0 - NoH * NoH;
+    float a = NoH * roughness;
+    float k = roughness / (oneMinusNoHSquared + a * a);
+    float d = k * k * (1.0 / PI);
+    return d;
+}
+
+float V_SmithGGXCorrelated(float roughness, float NoV, float NoL) {
+    float a2 = roughness * roughness;
+    float lambdaV = NoL * sqrt((NoV - a2 * NoV) * NoV + a2);
+    float lambdaL = NoV * sqrt((NoL - a2 * NoL) * NoL + a2);
+    float v = 0.5 / (lambdaV + lambdaL);
+    return v;
+}
+
+vec3 F_Schlick(const vec3 f0, float f90, float VoH) {
+    return f0 + (f90 - f0) * pow5(1.0 - VoH);
+}
+
+float F_Schlick(float f0, float f90, float VoH) {
+    return f0 + (f90 - f0) * pow5(1.0 - VoH);
```

Comment:
Is this approach faster than say `mix(f0, f90, pow5(1.0 - VoH));`?

Reply:
I don't know, but I agree it makes sense.

Reply:
Ah. The point is probably to have `F_Schlick` be the same for `float f0` and `vec3 f0`, the other version becomes this, which I'm not sure would be the performance impact of.

```
vec3 F_Schlick(const vec3 f0, float f90, float VoH) {
    return mix(f0, vec3(f90), pow5(1.0 - VoH));
}

float F_Schlick(float f0, float f90, float VoH) {
    return mix(f0, f90, pow5(1.0 - VoH));
}
```